justfile

```diff
@@ -0,0 +1,66 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+_default:
+    @just --list
+
+_env:
+    #!/usr/bin/env sh
+    if [ ! -f .env ]; then
+      echo "Copying docker/config/env-dist to .env..."
+      cp docker/config/env-dist .env
+    fi
+
+# Build docker images.
+build *args='base fakesentry statsd': _env
+    docker compose --progress plain build {{args}}
+
+# Run the webapp and services.
+run *args='--attach=eliot --attach=fakesentry eliot': _env
+    docker compose up {{args}}
+
+# Stop service containers.
+stop *args:
+    docker compose stop {{args}}
+
+# Remove service containers and networks.
+down *args:
+    docker compose down {{args}}
+
+# Open a shell in the web image.
+shell *args='/bin/bash': _env
+    docker compose run --rm --entrypoint= eliot {{args}}
+
+# Open a shell in the test container.
+test-shell *args='/bin/bash': _env
+    docker compose run --rm --entrypoint= test {{args}}
+
+# Stop and remove docker containers and artifacts.
+clean:
+    docker compose rm -f
```

Comment:
Running `docker compose rm` only removes containers, but not networks, containers and images. Should we rename `down` to `clean` instead?

Networks specifically can be a problem on Linux, since they create entries in the routing table that often conflict with random wifi networks, so I generally want `down` rather than just `rm` to clean things up. (The first time I ran into this was on a German train. The wifi on German trains uses a `172.x.0.0/16` network, and so does Docker, and they often conflict, with the result that you can connect to the wifi, but no IP packets are ever sent there. This wasn't easy to debug.)

Reply:
I think I should make `clean` do `docker compose down`, but I would like to keep `down` as a separate command, and add more things to `clean`, like this line from `docs`:
```
docker compose run --rm --no-deps eliot bash make -C docs/ clean
```